pkg/charts/linkerd2/tracing.go

```diff
@@ -0,0 +1,71 @@
+package linkerd2
+
+import (
+	"k8s.io/helm/pkg/chartutil"
+)
+
+type (
+
+	// Tracing consists of the add-on configuration of the distributed tracing components sub-chart.
+	Tracing struct {
+		Enabled   bool       `json:"enabled"`
+		Collector *Collector `json:"collector"`
+		Jaeger    *Jaeger    `json:"jaeger"`
+	}
+
+	// Collector consists of the config values required for Trace collector
+	Collector struct {
+		Name      string     `json:"name"`
+		Image     string     `json:"image"`
+		Resources *Resources `json:"resources"`
+	}
+
+	// Jaeger consists of the config values required for Jaeger
+	Jaeger struct {
+		Name      string     `json:"name"`
+		Image     string     `json:"image"`
+		Resources *Resources `json:"resources"`
+	}
+)
+
+var (
+	tracingChartName   = "tracing"
+	tracingConfigStage = []string{
+		"templates/tracing-rbac.yaml",
+	}
+
+	tracingControlPlaneStage = []string{
+		"templates/tracing.yaml",
+	}
+)
+
+// GetChartName returns the name of the add-on sub-chart
+func (*Tracing) GetChartName() string {
+	return tracingChartName
+}
+
+// GetFiles returns the templates files that are part of the add-on sub-chart
+func (*Tracing) GetFiles() []*chartutil.BufferedFile {
+	files := []*chartutil.BufferedFile{
+		{Name: chartutil.ChartfileName},
+	}
+
+	for _, template := range tracingConfigStage {
+		files = append(files, &chartutil.BufferedFile{
+			Name: template,
+		})
+	}
+
+	for _, template := range tracingControlPlaneStage {
+		files = append(files, &chartutil.BufferedFile{
+			Name: template,
+		})
+	}
+
+	return files
+}
```

Comment:
In order to avoid repeating this function into each add-on, WDYT about embedding into the `Tracing` struct another struct `BaseAddon` declared in `add-on.go` that would implement `GetFiles(configStage, controlPlaneStage []string) []*chartutil.BufferedFile` and so `GetFiles` can just be:
```go
func (tr *Tracing) GetFiles() []*chartutil.BufferedFile {
    return tr.BaseAddon.GetFiles(tracingConfigStage, tracingControlPlaneStage)
}
```

Reply:
Adding a field to the `tracing` struct, would unnecessarily add more complexity as we then decode that into a yaml for `linkerd-values`. Instead I added a `defaultGetFiles` function with your suggested signature :)